src/main/java/org/hibernate/infra/asciidoctor/extensions/customnumbering/CustomNumberingProcessor.java

```diff
@@ -37,39 +44,39 @@ public CustomNumberingProcessor(Map<String, Object> config) {
 	@Override
 	public Document process(Document document) {
 		document.getBlocks().stream()
-				.filter( abstractBlock -> "section".equalsIgnoreCase( abstractBlock.getNodeName() ) )
-				.forEach( section -> processSection( section, NumberingIndexes.get( getSectionNumber( section ) ) ) );
+				.filter( block -> block instanceof Section )
+				.forEach( block -> processSection( (Section) block, getSectionNumberingIndexes( getSectionNumber( (Section) block ) ) ) );
 		return document;
 	}
 
-	private void processSection(AbstractBlock section, NumberingIndexes indexes) {
+	private void processSection(Section section, SectionNumberingIndexes indexes) {
 		for ( AbstractBlock block : section.getBlocks() ) {
 			if ( "example".equalsIgnoreCase( block.getNodeName() ) ) {
```

Comment:
Note that there's unfortunately no `Example` type... It's probably something we should try to get improved in Asciidoctorj.